lib/http-server.js

```diff
@@ -54,13 +54,18 @@ function HttpServer(options) {
     options.cache === -1 ? 'no-cache, no-store, must-revalidate' :
     options.cache // in seconds.
   );
-  this.showDir = options.showDir !== 'false';
-  this.autoIndex = options.autoIndex !== 'false';
-  this.showDotfiles = options.showDotfiles;
-  this.gzip = options.gzip === true;
-  this.brotli = options.brotli === true;
+  if(options.showDir)
+    this.showDir = options.showDir.toLowerCase() !== 'false';
+  if(options.autoIndex)
+    this.autoIndex = options.autoIndex.toLowerCase() !== 'false';
+  if(options.showDotfiles)
+    this.showDotfiles = options.showDotfiles.toLowerCase() !== 'false';
+  if(options.gzip)
+    this.gzip = options.gzip.toLowerCase() !== 'false';
+  if(options.brotli)
+    this.brotli = options.brotli.toLowerCase() !== 'false';
   if (options.ext) {
-    this.ext = options.ext === true
+    this.ext = options.ext.toLowerCase() !== 'false'
```

Comment:
Since these are actual boolean options (rather than strings), we can't `.toLowerCase()` them, and I think we can safely leave them be in this PR

Reply:
Actually, I checked it out (`console.log`ed the typeof options)...they are strings

Reply:
for example, try `node ./bin/http-server ./public --gzip=false` in the current master branch code, this will parse gzip option as `true`

Reply:
Now that I look at the npm test results, `Error: TypeError: options.gzip.toLowerCase is not a function` I think I need to look at it a bit more

Reply:
Ok, found the issue:
`-g` returns a boolean
`-g false` returns a string
This is the cause of the inconsistency (adding a value after the flag)

Reply:
this is a feature of how the parsing library, `minimist` works, and I think we're fine leaving these as-is for now